modules/pam_unix/passverify.c

```diff
@@ -500,7 +500,7 @@ PAMH_ARG_DECL(char * create_password_hash,
 #else
 	sp = stpcpy(salt, algoid);
 	if (on(UNIX_ALGO_ROUNDS, ctrl)) {
-		sp += snprintf(sp, sizeof(salt) - (16 + 1 + (sp - salt)), "rounds=%u$", rounds);
+		sp += snprintf(sp, sizeof(salt) - (16 + 1 + (sp - salt)), "rounds=%i$", rounds);
```

Comment:
Given that `rounds < 0` doesn't make any sense, I suggest changing the type of `rounds` to an unsigned type instead.

Reply:
Changing `rounds` to an unsigned type will require changing the argument type in `_set_ctrl()` also called in `pam_sm_acct_mgmt()`, `pam_sm_authenticate(),` `pam_sm_setcred()`, `pam_sm_open_session()` and `pam_sm_close_session()` where `int argc` is passed.

Reply:
Anyway, printing `rounds` as a signed integer doesn't make sense.